src/main/user-impl/java/com/mysql/cj/jdbc/ha/ConnectionProxy.java

```diff
@@ -182,7 +187,7 @@ public synchronized Object invoke(Object proxy, Method method, Object[] args)
           this.currentConnection.getClass(),
           methodName,
           () -> method.invoke(currentConnection, args),
-          args);
+          Arrays.copyOf(args, args.length));
```

Comment:
I don't think we need to explicitly call `Arrays.copyOf` here. Modifying args shouldn't affect the args used in `method.invoke`